code/edgeidentity/src/main/java/com/adobe/marketing/mobile/edge/identity/IdentityStorageManager.java

```diff
@@ -0,0 +1,134 @@
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.edge.identity;
+
+import static com.adobe.marketing.mobile.edge.identity.IdentityConstants.LOG_TAG;
+
+import androidx.annotation.VisibleForTesting;
+import com.adobe.marketing.mobile.services.DataStoring;
+import com.adobe.marketing.mobile.services.Log;
+import com.adobe.marketing.mobile.services.NamedCollection;
+import com.adobe.marketing.mobile.services.ServiceProvider;
+import com.adobe.marketing.mobile.util.JSONUtils;
+import com.adobe.marketing.mobile.util.StringUtils;
+import java.util.Map;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * Manages persistence for this Identity extension
+ */
+class IdentityStorageManager {
+
+	private static final String LOG_SOURCE = "IdentityStorageService";
+	private final NamedCollection edgeIdentityStore;
+	private final NamedCollection directIdentityStore;
+
+	IdentityStorageManager() {
+		this(ServiceProvider.getInstance().getDataStoreService());
+	}
+
+	@VisibleForTesting
+	IdentityStorageManager(final DataStoring dataStoreService) {
+		this.edgeIdentityStore = dataStoreService.getNamedCollection(IdentityConstants.DataStoreKey.DATASTORE_NAME);
+		this.directIdentityStore =
+			dataStoreService.getNamedCollection(IdentityConstants.DataStoreKey.IDENTITY_DIRECT_DATASTORE_NAME);
+	}
+
+	/**
+	 * Loads identity properties from local storage, returns null if not found.
+	 *
+	 * @return properties stored in local storage if present, otherwise null.
+	 */
+	IdentityProperties loadPropertiesFromPersistence() {
+		if (edgeIdentityStore == null) {
+			Log.debug(
+				LOG_TAG,
+				LOG_SOURCE,
+				"EdgeIdentity named collection is null. Unable to load saved identity properties from persistence."
+			);
+			return null;
+		}
+		final String jsonString = edgeIdentityStore.getString(IdentityConstants.DataStoreKey.IDENTITY_PROPERTIES, null);
+
+		if (jsonString == null) {
+			Log.debug(
+				LOG_TAG,
+				LOG_SOURCE,
+				"No previous properties were stored in persistence. Current identity properties are null"
+			);
+			return null;
+		}
+
+		try {
+			final JSONObject jsonObject = new JSONObject(jsonString);
+			final Map<String, Object> propertyMap = JSONUtils.toMap(jsonObject);
+			return new IdentityProperties(propertyMap);
+		} catch (JSONException exception) {
+			Log.debug(
+				LOG_TAG,
+				LOG_SOURCE,
+				"Serialization error while reading properties jsonString from persistence. Unable to load saved identity properties from persistence."
+			);
+			return null;
+		}
+	}
+
+	/**
+	 * Saves identity properties to local storage.
+	 *
+	 * @param properties properties to be stored
+	 */
+	void savePropertiesToPersistence(final IdentityProperties properties) {
+		if (edgeIdentityStore == null) {
+			Log.debug(
```

Comment:
unexpected and the extension works in limited capacity, let's add a warning for this